internal/driver/gomobile/file_android.go

```diff
@@ -92,6 +94,50 @@ func nativeFileOpen(f *fileOpen) (io.ReadCloser, error) {
 	return stream, nil
 }
 
+func saveStream(uri string) unsafe.Pointer {
+	uriStr := C.CString(uri)
+	defer C.free(unsafe.Pointer(uriStr))
+
+	var stream unsafe.Pointer
+	app.RunOnJVM(func(_, env, ctx uintptr) error {
+		streamPtr := C.saveStream(C.uintptr_t(env), C.uintptr_t(ctx), uriStr)
+		if streamPtr == C.NULL {
+			return os.ErrNotExist
+		}
+
+		stream = unsafe.Pointer(streamPtr)
+		return nil
+	})
+	return stream
+}
+
+func nativeFileSave(f *fileSave) (io.WriteCloser, error) {
+	if f.uri == nil || f.uri.String() == "" {
+		return nil, nil
+	}
+
+	ret := saveStream(f.uri.String())
+	if ret == nil {
+		return nil, errors.New("resource not found at URI")
+	}
+
+	stream := &javaStream{}
+	stream.stream = ret
+	return stream, nil
+}
+
+// Declare conformity to WriteCloser interface
+var _ io.WriteCloser = (*javaStream)(nil)
```

Comment:
this assertion?

Reply:
Apologies, I was away from computer and could not find it. I mean this:
https://github.com/fyne-io/fyne/blob/develop/internal/driver/gomobile/repository.go#L16